File: AdaptiveLayout/before/AdaptiveUI/app/src/main/java/bawo/adaptiveui/MainActivity.java
package bawo.adaptiveui;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import bawo.adaptiveui.adapter.PostAdapter;
import bawo.adaptiveui.data.PostData;

public class MainActivity extends AppCompatActivity {
    private RecyclerView recyclerView;
    private RecyclerView.Adapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        adapter = new PostAdapter(this, PostData.postList());
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);

    }
}

File: KotlinForAndroid/android/SimpleList2ListView/app/src/main/java/randombgappkotlinandroid/gohool/com/simplelist2_listview/MainActivity.kt
package randombgappkotlinandroid.gohool.com.simplelist2_listview

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    private var adapter: PersonListAdapter? = null
    private var personList: ArrayList<Person>? = null
    private var layoutManager: RecyclerView.LayoutManager? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        personList = ArrayList<Person>()
        layoutManager = LinearLayoutManager(this, LinearLayout.VERTICAL, false) as RecyclerView.LayoutManager?
        adapter = PersonListAdapter(personList!!, this)


        //setup list
        recyclerviewId.layoutManager = layoutManager
        recyclerviewId.adapter = adapter


        //load data
        for (i in 0..9) {
            val mPerson = Person()
            mPerson.name = "Paulo_" + i
            mPerson.age = 12 + i
            personList!!.add(mPerson)
        }
        adapter!!.notifyDataSetChanged()




    }
}
